watchtower/__init__.py

```diff
@@ -9,23 +11,46 @@
 from botocore.exceptions import ClientError
 
 
-def _idempotent_create(client, method, *args, **kwargs):
+if TYPE_CHECKING:
+    from typing_extensions import Protocol
+
+    class LogClient(Protocol):
+        exceptions: Any
+
+        def create_log_group(self, logGroupName: str) -> None:
+            ...
+
+        def create_log_stream(self, logGroupName: str, logStreamName: str) -> None:
+            ...
+
+        def put_log_events(self, **kwargs: Any) -> dict:
+            ...
+
+        def put_retention_policy(self, logGroupName: str, retentionInDays: Optional[int]) -> None:
+            ...
+
+    class Session(Protocol):
+        def client(self, service: str, endpoint_url: Optional[str] = None) -> LogClient:
+            ...
```

Comment:
If these are too verbose they can be left as `Any`. I'm not sure if mypy understands how to follow `_idempotent_create` so `LogClient` may only need to provide `put_log_events`, but I declared them all since their are brief and it might be more surprising if `_idempotent_create` does get followed and then there are complaints that the methods don't exist.